File: daily_python/1752_Check_if_Array_Is_Sorted_and_Rotated.py
import os
import sys
from typing import List

class Solution:
    def check(self, nums: List[int]) -> bool:
        inverse = 0
        for i in range(1, len(nums)):
            if nums[i-1] > nums[i]:
                inverse += 1
        if nums[len(nums)-1] > nums[0]:
            inverse += 1
        return inverse <= 1
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.check(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/1752_Check_if_Array_Is_Sorted_and_Rotated.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        int r = 0;
        for (int i = 0; i < n; i++){
            if (nums[i] > nums[(i+1) % n]){
                r++;
                if (r > 1) return false;
            }
        }
        return true;
    }
};

int main(){
    vector<int> nums = {3,4,5,1,2};
    // vector<int> nums = {2,1,3,4};
    // vector<int> nums = {1,2,3};
    // vector<int> nums = {6,10,6};
    Solution S;
    cout << S.check(nums) << endl;
    return 0;
}